From d1309c8d531ee0d96089dd54b326f18b6cab5f4f Mon Sep 17 00:00:00 2001
From: Juergen Repp <juergen_repp@web.de>
Date: Sat, 26 Apr 2025 13:36:32 +0200
Subject: [PATCH 16/18] tools: Add check whether fopen fails.

In tpm2_commit.c and tpm2_ecephemeral.c the check whether the
result of fopen is NULL is added.

Signed-off-by: Juergen Repp <juergen_repp@web.de>
---
 tools/tpm2_commit.c      | 7 +++++++
 tools/tpm2_ecephemeral.c | 7 +++++++
 2 files changed, 14 insertions(+)

diff --git a/tools/tpm2_commit.c b/tools/tpm2_commit.c
index 180eeb91..cfe43d8e 100644
--- a/tools/tpm2_commit.c
+++ b/tools/tpm2_commit.c
@@ -1,5 +1,6 @@
 /* SPDX-License-Identifier: BSD-3-Clause */
 #include <string.h>
+#include <errno.h>
 
 #include "files.h"
 #include "log.h"
@@ -100,6 +101,12 @@ static tool_rc process_outputs(ESYS_CONTEXT *ectx) {
     }
 
     FILE *fp = fopen(ctx.commit_counter_path, "wb");
+    if (!fp) {
+        LOG_ERR("Could not open output file \"%s\" error: \"%s\"",
+                ctx.commit_counter_path, strerror(errno));
+        return tool_rc_general_error;
+    }
+    
     is_file_op_success = files_write_16(fp, ctx.counter);
     fclose(fp);
     if (!is_file_op_success) {
diff --git a/tools/tpm2_ecephemeral.c b/tools/tpm2_ecephemeral.c
index 8783db7f..5c840803 100644
--- a/tools/tpm2_ecephemeral.c
+++ b/tools/tpm2_ecephemeral.c
@@ -1,5 +1,7 @@
 /* SPDX-License-Identifier: BSD-3-Clause */
 
+#include <errno.h>
+
 #include "files.h"
 #include "log.h"
 #include "tpm2.h"
@@ -74,6 +76,11 @@ static tool_rc process_outputs(ESYS_CONTEXT *ectx) {
      * 2. Outputs generated after TPM2_CC_<command> dispatch
      */
     FILE *fp = fopen(ctx.commit_counter_path, "wb");
+    if (!fp) {
+        LOG_ERR("Could not open output file \"%s\" error: \"%s\"",
+                ctx.commit_counter_path, strerror(errno));
+        return tool_rc_general_error;
+    }
     is_file_op_success = files_write_16(fp, ctx.counter);
     fclose(fp);
     if (!is_file_op_success) {
-- 
2.49.0

